src/cmakeTools.ts

```diff
@@ -1799,8 +1799,7 @@ export class CMakeTools implements api.CMakeToolsAPI {
         }
         await this.setDefaultBuildTarget(target);
         const drv = await this.cmakeDriver;
-        const targets = await this.getDefaultBuildTargets();
-        this.updateDriverAndTargetsInTaskProvider(drv, targets);
+        this.updateDriverAndTargetsInTaskProvider(drv, [target]);
```

Comment:
Revert this for now.  I think that we actually shouldn't set anything here. Instead we should be updating the task provider when `configure` completes because we should have the list of available targets at that point.